src/ReturnTypes/RequestExtension.php

```diff
@@ -41,6 +45,16 @@ public function getTypeFromMethodCall(
         MethodCall $methodCall,
         Scope $scope
     ): Type {
-        return new MixedType;
+        $uploadedFileType = new ObjectType(UploadedFile::class);
+
+        if (count($methodCall->args) === 0) {
+            return new ArrayType(new IntegerType(), $uploadedFileType);
+        }
+
+        if (count($methodCall->args) === 1) {
+            return TypeCombinator::addNull($uploadedFileType);
```

Comment:
I don't think this is correct.

```php
$request->file('documents')
```
may return an array as well. It depends on the request.

If the request has the keys: `documents[0]` and `documents[1]` then the `$request->file('documents')` returns an array.

Reply:
Didn't know that. Good point. Thank you :+1:

Reply:
Updated :+1: